srcs/app/config/analyser/ParserStatelessSet.cpp

```diff
@@ -0,0 +1,303 @@
+		throw SyntaxError("Expecting `on' or `off' but found `" +
+		data.GetRawData()  + "'", LINE);
+	config_->AddAutoindex(data.GetRawData() == "on", data.GetCtx(), LINE);
+	return Parser::State::K_EXP_SEMIC;
+}
+
+t_parsing_state Parser::StatelessSet::ServerNameHandlerSemic
+(const StatefulSet &data) {
+	if (data.GetArgNumber() == 0)
+		throw Analyser::SyntaxError("Invalid number of arguments in "
+									"`server_name' directive", LINE);
+	config_->AddServerName(parser_->GetArgs(), data.GetCtx(), LINE);
+	parser_->ResetArgNumber();
+	return Parser::State::K_EXP_KW;
+}
+
+t_parsing_state Parser::StatelessSet::ServerNameHandler
+													(const StatefulSet &data) {
+	parser_->IncrementArgNumber(data.GetRawData());
+	return Parser::State::K_SERVER_NAME;
+}
+
+bool Parser::StatelessSet::areHttpMethodsValid_(const std::vector<std::string>
+										&input, std::string *error_throw) {
+	for (unsigned int i = 0; i < input.size(); ++i) {
+		bool notValidMethod = true;
+		for (unsigned j = 0;
+			 j < sizeof(valid_http_methods)/sizeof(valid_http_methods[0]);
+			 ++j )
+			notValidMethod = notValidMethod
+				&& (valid_http_methods[j] != input[i]);
+		if (notValidMethod) {
+			*error_throw = "`" + input[i] + "' is not"
+				" a valid http method for `limit_except' directive";
+			return false;
+		}
+	}
+	return true;
+}
+
+
+t_parsing_state Parser::StatelessSet::LimitExceptHandlerSemic
+(const StatefulSet &data) {
+	std::string error_throw;
+	if (data.GetArgNumber() == 0)
+		throw Analyser::SyntaxError("Invalid number of arguments in "
+									"`limit_except' directive", LINE);
+	if (!areHttpMethodsValid_(parser_->GetArgs(), &error_throw)) {
+		throw Analyser::SyntaxError(error_throw, LINE);
+	}
+	config_->AddLimitExcept(parser_->GetArgs(), data.GetCtx(), LINE);
+	parser_->ResetArgNumber();
+	return Parser::State::K_EXP_KW;
+}
+
+
+t_parsing_state Parser::StatelessSet::LimitExceptHandler
+(const StatefulSet &data) {
+	parser_->IncrementArgNumber(data.GetRawData());
+	return Parser::State::K_LIMIT_EXCEPT;
+}
+
+t_parsing_state Parser::StatelessSet::CgiAssignHandler
+													(const StatefulSet &data) {
+	if (data.GetArgNumber() == 0) {
+		parser_->IncrementArgNumber(data.GetRawData());
+		return Parser::State::K_CGI_ASSIGN;
+	} else if (data.GetArgNumber() == 1) {
+		config_->AddCgiAssign(parser_->GetArgs()[0],
+							  data.GetRawData(),
+							  data.GetCtx(),
+							  LINE);
+		parser_->ResetArgNumber();
+		return Parser::State::K_EXP_SEMIC;
+	} else {
+		throw Analyser::SyntaxError("Invalid number of arguments in "
+									"`cgi_assign' directive", LINE);
+	}
+}
+
+bool Parser::StatelessSet::isReturnStatusValid_(int64_t status) {
+	for (size_t i = 0;
+		 i < sizeof(valid_return_status)/sizeof(valid_return_status[0]);
+		 ++i) {
+		if (static_cast<uint16_t>(status) == valid_return_status[i]) {
+			return true;
+		}
+	}
+	return false;
+}
+
+t_parsing_state Parser::StatelessSet::ReturnHandler
```

Comment:
To allow only HTTP redirections, we could improve the return URI validation. It should start with `http://` or `https://` and have at least one more character after.

Reply:
done!